### DISCUSSION TITLE
Capture query string in routed functions

### DISCUSSION THREAD

**Discussion by ZeeD** [paveldedik/ludic#91] [General]
I'm having troubles handling some query parameters in some routes
let's say I want to handle http://www.example.com/foo?bar=baz , how can I read the `bar` value?
Ideally I would have loved to write just

```python
@app.get('/foo')
def foo(*, bar: str|None) -> ...:
    ...
```

but it seems that ludic doesn't support this style, as all I got is an Internal Server Error...

I can write

```python
@app.get('/foo')
def foo(request: Request) -> ...:
    bar: str | None = request.query_params.get('bar')
    ...
```

but it seems a bit contrived (and I would have liked some automatic validation and/or conversion to int/bool types), yet I've searched in the docs but I didn't found any other way..

**Comment by paveldedik** [comment]
Hi, thank you for the question. You are right that the second option is currently the only way to do it. But I like your first proposal and I think I'll implement it in that or similar way (If there is no problem). Also a good to know there is an internal server error currently for that view, I definitely need to fix that.

I'll look into that soon, I like your suggestion.

**Reply by paveldedik** [reply]
Hi @ZeeD 

This should be working now (for 0.5.3 and above):

```python
@app.get('/foo')
def foo(*, bar: str | None) -> ...:
    ...
```

Not that you can also do the following (I didn't mention it before):

```python
from ludic.web.datastructures import QueryParams

@app.get('/foo')
def foo(query_params: QueryParams) -> ...:
    bar_value = query_params.get("bar")
```

**Reply by ZeeD** [reply]
thank you so much!!
